src/core/ext/xds/xds_cluster.h

```diff
@@ -101,7 +101,9 @@ struct XdsClusterResource : public XdsResourceType::ResourceData {
 
   XdsHealthStatusSet override_host_statuses;
 
-  std::shared_ptr<std::map<std::string, std::string>> telemetry_labels;
+  std::vector<std::pair<ClientCallTracer::CallAttemptTracer::OptionalLabelKey,
```

Comment:
Can this be a `std::array<RefCountedStringValue, ClientCallTracer::CallAttemptTracer::OptionalLabelKey::kSize>` since we know the upper-bound of the size in compile-time?

Reply:
The main benefit for std::vector is the contiguous storage of elements so that when we iterate over elements for a call, it'll be faster compared to an std::map.

But, as you said, for a small number of elements, the performance is not gonna matter much, so if you feel strongly about this, I'll revert to an `std::map`.

Reply:
I would prefer not having an API that allows misuse like duplicate labels.  An alternative would be to just have two fields here, one for each label:

```
RefCountedStringValue service_label;
RefCountedStringValue namespace_label;
```

This avoids all container overhead, with the trade-off that we spend memory for two pointers even if the labels are not present.

Reply:
The size of `std::map<>` is 48 vs `RefCountedStringValue` which is 8